### PR TITLE
✅ Add Preact Enzyme tests

### PR DESCRIPTION
**Pull request by jridgewell**

### CODE REVIEW

File: extensions/amp-date-display/0.2/test/test-amp-date-display.js [hunk 1]
```diff
@@ -112,135 +109,20 @@ describes.realWin(
...
-        element.build();
-
-        const data = await getRenderedData();
-
-        expect(data.iso).to.equal('2001-02-03T04:05:06.007Z');
-      });
-    });
-
-    it('adds offset seconds', async () => {
+    it('does not rerender', async () => {
```
Comment (caroqliu) [inline extensions/amp-date-display/0.2/test/test-amp-date-display.js]: This is because `amp-date-display` doesn't interact with `amp-bind` at all, right? Why is this needed? Custom elements that don't use a callback typically also don't explicitly check that it's ineffective if called.

File: extensions/amp-date-display/0.2/test/test-date-display.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+  beforeEach(() => {
+    sandbox = env.sandbox;
+    sandbox.useFakeTimers(new Date('2018-01-01T08:00:00Z'));
+  });
+
+  // Unfortunately, we cannot test the most interesting case of UTC datetime
+  // displayed in local, because the test would work in only one time zone.
+
+  it('provides all variables in UTC and English (default)', () => {
+    const props = {
```
Comment (caroqliu) [inline extensions/amp-date-display/0.2/test/test-date-display.js]: Why not create a helper method that takes `props` and returns `data` for all these test cases?

File: extensions/amp-date-display/0.2/test/test-amp-date-display.js (hunk 1 above)
Comment (jridgewell) [reply extensions/amp-date-display/0.2/test/test-amp-date-display.js]: It's to keep the same API as the original `amp-date-display`. It only uses `buildCallback`, which is just the first "render" called here.

File: extensions/amp-date-display/0.2/test/test-date-display.js (hunk 2 above)
Comment (jridgewell) [reply extensions/amp-date-display/0.2/test/test-date-display.js]: go/tott/598. Basically, don't rely on test logic when it's not necessary.
